web/src/main/java/org/springframework/security/web/authentication/www/AuthenticationTokenConverter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.web.authentication.www;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.authentication.AuthenticationDetailsSource;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+
+/**
+ * Decorates an underlying {@link AuthenticationConverter} with common
+ * logic needed for {@link AbstractAuthenticationToken}.
+ *
+ * @author Sergey Bespalov
+ *
+ * @param <T> target {@link AbstractAuthenticationToken} type
+ */
+public class AuthenticationTokenConverter<T extends AbstractAuthenticationToken> implements AuthenticationConverter<T> {
```

Comment:
Let's remove this. I realize this removes some duplication within the converters, but I prefer the minimal duplication (most the work is done in the AuthenticationDetailsSource) over the complexity of setting things up properly.

Reply:
fixed